### PR TITLE
i#2272 drsyms uses the wrong criteria to distinguish imports on aarch64

### PR DESCRIPTION
**Pull request by Francesc-Martinez**
Fixed the logic used to detect imports on ext/drsyms/drsyms_elf.c.
Now we use the same logic as in core/unix/module_elf.c symbol_is_import.

Fixes issue #2272

### CODE REVIEW

**Comment by egrimley** [comment]
The Travis failure looks like i#2003.

File: ext/drsyms/drsyms_elf.c [hunk 1]
```diff
@@ -352,7 +354,9 @@ drsym_obj_symbol_offs(void *mod_in, uint idx, size_t *offs_start OUT,
     elf_info_t *mod = (elf_info_t *) mod_in;
     if (offs_start == NULL || mod == NULL || idx >= mod->num_syms || mod->syms == NULL)
         return DRSYM_ERROR_INVALID_PARAMETER;
-    if (mod->syms[idx].st_value == 0) {
+    if ((mod->syms[idx].st_value == 0 &&
+        ELF_ST_TYPE(mod->syms[idx].st_info) != STT_TLS) ||
+        mod->syms[idx].st_shndx == 0) {
```
Comment (derekbruening) [inline ext/drsyms/drsyms_elf.c]: STN_UNDEF would be nicer but requires defining for some android toolchains.

File: ext/drsyms/drsyms_elf.c [hunk 2]
```diff
@@ -352,7 +354,9 @@ drsym_obj_symbol_offs(void *mod_in, uint idx, size_t *offs_start OUT,
     elf_info_t *mod = (elf_info_t *) mod_in;
     if (offs_start == NULL || mod == NULL || idx >= mod->num_syms || mod->syms == NULL)
         return DRSYM_ERROR_INVALID_PARAMETER;
-    if (mod->syms[idx].st_value == 0) {
+    if ((mod->syms[idx].st_value == 0 &&
```
Comment (derekbruening) [inline ext/drsyms/drsyms_elf.c]: suggest: a comment asking to keep consistent with symbol_is_import() in the core

Comment (derekbruening) [inline ext/drsyms/drsyms_elf.c]: Plus a comment in module_elf.c pointing back here, so if we change one we will change the other (ideally we would share the code: would have to be some API routine, unclear if that makes sense though).

Comment (Francesc-Martinez) [inline ext/drsyms/drsyms_elf.c]: Ok, I added/modified the comments to keep the code synchronized.

File: ext/drsyms/drsyms_elf.c (hunk 1 above)
Comment (Francesc-Martinez) [inline ext/drsyms/drsyms_elf.c]: I can add it, but I thought it would be a bit too much to add another #ifndef/#define/#endif to use it only one time.

**Comment by derekbruening** [comment]
Thanks again for the fix.

**Pull request merged**